src/components/Button/index.js

```diff
@@ -1,252 +1,111 @@
+  color: ${(props) => (props.color ? props.color : `#ffffff`)};
+  background-color: ${(props) =>
+    props.background ? props.background : `#dc3545`};
+  font-weight: 400;
+  text-align: center;
+  white-space: nowrap;
+  vertical-align: middle;
+  border: 1px solid transparent;
+  padding: .375rem .75rem;
+  font-size: 1rem;
+  line-height: 1.5;
+  border-radius: .25rem;
+  transition: color .15s ease-in-out,background-color .15s ease-in-out,border-color .15s ease-in-out,box-shadow .15s ease-in-out;
+}`;
 
-// The button styled component
-const StyledButton = styled.button`
-  font-style: normal;
-  font-weight: 600;
-  font-size: ${(props) => props.variant.fontSize || "14px"};
-  font-family: "Open Sans", sans-serif;
-  line-height: 14px;
-  color: ${(props) => props.variant.color};
-  display: flex;
-  align-items: center;
+export const StyledOutlinedButton = styled.button`
+  width: 135.88px;
+  left: 2354px;
+  top: 1432px;
+  color: ${(props) => (props.color ? props.color : `#007bff`)};
+  background-color: ${(props) =>
+    props.background ? props.background : `#ffffff`};
+  font-weight: 400;
   text-align: center;
-  letter-spacing: 0.75px;
-  text-transform: capitalize;
-  padding: ${(props) => props.variant.padding || "8px 32px"};
-  box-shadow: ${(props) => props.variant.boxShadow || "none"};
-  border: ${(props) => props.variant.border || "none"};
-  background: ${(props) => props.variant.background};
-  opacity: ${(props) => props.variant.opacity || "initial"};
-  border-radius: ${(props) => props.variant.borderRadius || "5px"};
+  white-space: nowrap;
+  vertical-align: middle;
+  border: 1px solid ${(props) => (props.color ? props.color : `#007bff`)};
+  padding: 0.375rem 0.75rem;
+  font-size: 1rem;
+  line-height: 1.5;
+  border-radius: 0.25rem;
+  transition: color 0.15s ease-in-out, background-color 0.15s ease-in-out,
+    border-color 0.15s ease-in-out, box-shadow 0.15s ease-in-out;
 `;
 
-const Button = ({ variant, children, onClick }) => (
-  <StyledButton onClick={onClick} variant={getSelectedButtonStyle(variant)}>
-    {children}
-  </StyledButton>
+export const PrimaryButton = ({ ...args }) => (
+  <StyledPrimaryButton background={args.backgroundColor} color={args.color}>
+    {args.label}
+  </StyledPrimaryButton>
 );
 
-Button.propTypes = {
-  children: PropTypes.any.isRequired,
-  variant: PropTypes.string.isRequired,
-  onClick: PropTypes.func.isRequired,
-};
+export const SecondaryButton = ({ ...args }) => (
+  <StyledSecondaryButton background={args.backgroundColor} color={args.color}>
+    {args.label}
+  </StyledSecondaryButton>
+);
+
+export const DangerButton = ({ ...args }) => (
+  <StyledDangerButton background={args.backgroundColor} color={args.color}>
+    {args.label}
+  </StyledDangerButton>
+);
 
-export default Button;
+export const OutlinedButton = ({ ...args }) => (
+  <StyledOutlinedButton background={args.backgroundColor} color={args.color}>
+    {args.label}
+  </StyledOutlinedButton>
+);
```

Comment:
I like the implementation, but as we are using styled components you can export a single Button component and accept a property called, for instance, "variant" which cant take 4 different values ('primary', 'secondary', 'danger', 'outlined'), so then you can use the button as so 
```
<Button variant="primary>
   Click Here
</Button>
```

Inside the component you can then change implementation for each type as such
```
export const Button = styled.button`
  ${({ variant }) => css`
      color: #ffffff;
      background-color: #dc3545;
      font-weight: 400;
      text-align: center;
  `}
  top: 1432px;
  color: #ffffff;
  background-color: #dc3545;
  font-weight: 400;
  text-align: center;
  white-space: nowrap;
  vertical-align: middle;
   
   ${variant === 'secondary' && css`
       // Here the style that differs in the secondary
   `}

    ${variant === 'danger' && css`
       // Here the style that differ in the secondary
   `}

}`;
```

Reply:
By doing so, you will not have to replicate common styles between different buttons, and also you will be able to export a single component where user can change the style by simply updating the value of variant

Reply:
Thanks Davidec, really useful stuff. I made the changes as you advised. Thanks

Reply:
I refactored the code . Removed controllers and cleaned unnecessary parts.